packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -177,7 +171,23 @@ export const SelectionHandler = (
     isLeftClick = lastDownEvent.button === 0;
   };
 
+  // Helper
+  const upsertCurrentTarget = () => {
+    const exists = store.getAnnotation(currentTarget.annotation);
+    if (exists) {
+      store.updateTarget(currentTarget);
+    } else {
+      store.addAnnotation({
+        id: currentTarget.annotation,
+        bodies: [],
+        target: currentTarget
+      });
+    }
+  }
+
   const onPointerUp = (evt: PointerEvent) => {
+    if (isContextMenuOpen) return;
```

Comment:
Could you please clarify why the context menu guard was added for the `pointer` event handlers? With it in place, it takes 2 taps to dismiss the annotation selection. The first tap is needed to dismiss the context menu and the second one is to dismiss the annotation itself.

However, the annotation popup gets dismissed on the first tap too due to the `outside-press` reason! 
![image](https://github.com/user-attachments/assets/08520622-5297-4b3f-b4a3-80410be9c160)

That makes the annotation selection look like it's lagging behind 🤔 

https://github.com/user-attachments/assets/d9e3526c-dedb-4582-9a3c-a055968e79a3